crates/runtime/src/model/eval/mod.rs

```diff
@@ -193,6 +213,22 @@ async fn run_eval(
         vec![]
     };
 
+    if actual.len() != ideal.len() {
```

Comment:
The Snafu `ensure` macro can make this a bit nicer: https://docs.rs/snafu/latest/snafu/macro.ensure.html